src/handlers/movies/getMovies.ts

```diff
@@ -0,0 +1,79 @@
+import { getAll as getAllMovies } from '../../models/Movie';
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import getAll from '../../models/Genre/getAll';
+
+export const handler: APIGatewayProxyHandler = async (event) => {
+	try {
+		const queryParams: {
+			page?: number;
+			limit?: number;
+			genre?: string;
+			search?: string;
+			ids?: string[];
+		} = {};
+
+		const { queryStringParameters } = event;
+
+		if (queryStringParameters) {
+			const { page, limit, genre, search } = queryStringParameters;
+
+			if (page) {
+				const parsedPage = parseInt(page, 10);
+				if (!isNaN(parsedPage) && parsedPage > 0) {
+					queryParams.page = parsedPage;
+				}
+			}
+
+			if (limit) {
+				const parsedLimit = parseInt(limit, 10);
+				if (!isNaN(parsedLimit) && parsedLimit > 0) {
+					queryParams.limit = parsedLimit;
+				}
+			}
+
+			if (genre) {
+				queryParams.genre = genre;
+			}
+
+			if (search) {
+				queryParams.search = search;
+			}
+		}
+
+		if (queryParams.genre) {
+			// 	we need to use the other getAll function because we need to filter by genre and Contentful doesn't support filtering by reference fields many to many
+			const moviesWithGenreEntries = await getAll({
+				include: 1,
+				query: {
+					title: queryParams.genre,
+				},
+			});
+
+			const moviesIds = moviesWithGenreEntries.data[0].movies.map((movie) => movie.id);
+
+			queryParams.ids = moviesIds;
```

Comment:
If you're calling genre `getAll` with `include: 1` the list of movies returned by the Contentful API query already brings all data from the Movie content type. 

We could modify `parseGenre()` function to complete the fields you need, and then the call below `getAllMovies` wouldn't be necessary for this scenario

Reply:
This also needs to consider pagination. and sort

Reply:
I see 2 possibilities here:
1. keep the `include: 1` and sort and paginate `movies` returned by the genre API
2. change to `include: 0` (or omit this prop), and keep the second call `getAllMovies`

I'm leaning a bit more to the 1st option because it eliminates the necessity of the second call, but then you add more logic for the sort and pagination. Performance wise the 1st call is always better.

Reply:
The problem here is that when you call `getAll` from the Genre Model, it returns all the movies of that genre without taking into account the pagination passed. The other thing is that we couldn't sort by movies because Contentful doesn't allow us to filter by many to many references. I tried this, trying to avoid making two calls, but no luck! 

Good catch on the 2nd point though! I will let it as `include: 0`. Let me know your thoughts! Maybe I'm missing something here

Reply:
The filter, sort, and pagination would be done in the code since the API can't be done on Contentful's side.
```
// filter by search if received from call
const filteredMovies = moviesWithGenreEntries.data[0].movies.filter(...);

// sort for output and prepare pagination - maybe use sort from lodash?
const sortedMovies = sort(filteredMovies, 'title');

// apply pagination
const result = sortedMovies.slice((page - 1) * limit, page * limit); // double check if this is the right slice
```